homeassistant/components/modbus/sensor.py

```diff
@@ -141,11 +143,11 @@ async def async_setup_platform(
             _LOGGER.error("Error in sensor %s structure: %s", entry[CONF_NAME], err)
             continue
 
-        if entry[CONF_COUNT] * 2 != size:
+        if entry[CONF_DATA_SIZE] != size:
```

Comment:
That will not work if count is > 1.

Reply:
Work.

I add default = 2

Should do the same if dont use the data_size

Reply:
No it doesn’t because you removed the count check, so what happens if I specify count: 5

you need something like register_size * count != size